pkg/compute/deploy.go

```diff
@@ -84,11 +84,26 @@ func (c *DeployCommand) Exec(in io.Reader, out io.Writer) (err error) {
 
 	if c.version == 0 {
 		progress.Step("Fetching latest version...")
-		version, err := c.Globals.Client.LatestVersion(&fastly.LatestVersionInput{
+		versions, err := c.Globals.Client.ListVersions(&fastly.ListVersionsInput{
 			Service: serviceID,
 		})
 		if err != nil {
-			return fmt.Errorf("error getting latest service version: %w", err)
+			return fmt.Errorf("error getting listing service versions: %w", err)
+		}
+
+		// Get the latest non-locked version.
```

Comment:
Since versions are (deceivingly) non-linear, we might want to reconsider this approach. ie. If I have three versions:

- version 1: never activated
- version 2: cloned from 1, never activated
- version 3: cloned from 1, locked due to prior activation, currently inactive

... then choosing version 2 is not a good choice since it may be missing configuration from version 3. I think the right choice in this case is to clone version 3, though it may be difficult to ascertain. Leaving this note here for now as I investigate further.

Reply:
Couple options 

a) Treat active version as the thing we should always clone from , regardless of any future clones to avoid the problem Thom mentioned.  
b) if no active version, take highest inactive version that we can write to and use that as the version to deploy to.

Reply:
This PR implements (b) but I think we want to clone the most recently updated locked version instead. If there are no locked versions then we should use the highest version number. Certainly open other options!

Reply:
I see I think that makes sense.  I am not sure how besides activating you get the config into a state of "locked" though but if locked is the thing we want to clone from, +1 from me.

Reply:
Right, I feel like we can address both of your comments if revert back to the original logic and also clone the latest version if locked. I.e.

```suggestion
if c.version == 0 {
		progress.Step("Fetching latest version...")
		version, err := c.Globals.Client.LatestVersion(&fastly.LatestVersionInput{
			Service: serviceID,
		})
		if err != nil {
			return fmt.Errorf("error getting latest service version: %w", err)
		}

		if version.Active || version.Locked {
			progress.Step("Cloning latest version...")
			version, err = c.Globals.Client.CloneVersion(&fastly.CloneVersionInput{
				Service: serviceID,
				Version: version.Number,
			})
			if err != nil {
				return fmt.Errorf("error cloning latest service version: %w", err)
			}
		}

		c.version = version.Number
	}
```

Reply:
I spoke with Patrick via DM about this implementation. He has a path forward that will:

1. Choose the active version and clone it
2. If no active version exists, choose the latest updated locked version and clone that
3. If no locked version exists, choose the latest updated version and use it

Reply:
✅

Reply:
Have implemented the above in https://github.com/fastly/cli/commit/41af96c5c385101ba61942ad471189efacb9f2b7